### PR TITLE
Ensure defaults for float/double generation result in random values

### PR DESCRIPTION
**Pull request by niloc132**
Also ensures generating BigIntegers with a default range can contain some nulls.

Nightlies are currently running at https://github.com/niloc132/deephaven-core/actions/runs/6671871431

### CODE REVIEW

**Review by rcaudy — APPROVED** [review]
I like it enough, if Cristian does.

**Review by jcferretti — APPROVED** [review]
Maybe also remove that "1" in the constructor we discussed about, has no value/place in there.

**Comment by niloc132** [comment]
One more set of nightly runs with the better default lower bound https://github.com/niloc132/deephaven-core/actions/runs/6710144326

**Review by rcaudy — APPROVED** [review]

File: engine/test-utils/src/main/java/io/deephaven/engine/testutil/generator/DoubleGenerator.java [hunk 1]
```diff
@@ -22,7 +22,7 @@ public class DoubleGenerator extends AbstractGenerator<Double> {
     private final double posInfFraction;
 
     public DoubleGenerator() {
-        this(QueryConstants.NULL_DOUBLE + 1, Double.MAX_VALUE);
+        this(Math.nextUp(-Double.MAX_VALUE), Double.MAX_VALUE);
```
Comment (rcaudy) [inline engine/test-utils/src/main/java/io/deephaven/engine/testutil/generator/DoubleGenerator.java:25]: By the way, we have a constant for this:
`QueryConstants.MIN_FINITE_DOUBLE`

**Review by rcaudy — APPROVED** [review]

**Pull request merged**